docs/migration/cosmoshub-3.md

```diff
@@ -0,0 +1,190 @@
+double-signing and being slashed. The most important piece of this procedure is verifying your
+software version and genesis file hash before starting your validator and signing.
+
+The riskiest thing a validator can do is discover that they made a mistake and repeat the upgrade
+procedure again during the network startup. If you discover a mistake in the process, the best thing
+to do is wait for the network to start before correcting it. If the network is halted and you have
+started with a different genesis file than the expected one, seek advice from a Tendermint developer
+before resetting your validator.
+
+## Recovery
+
+Prior to exporting `cosmoshub-3` state, validators are encouraged to take a full data snapshot at the
+export height before proceeding. Snapshotting depends heavily on infrastructure, but generally this
+can be done by backing up the `.gaiacli` and `.gaiad` directories.
+
+It is critically important to back-up the `.gaiad/data/priv_validator_state.json` file after stopping your gaiad process. This file is updated every block as your validator participates in a consensus rounds. It is a critical file needed to prevent double-signing, in case the upgrade fails and the previous chain needs to be restarted.
+
+In the event that the upgrade does not succeed, validators and operators must downgrade back to
+gaia v2.0.14 with v0.37.15 of the _Cosmos SDK_ and restore to their latest snapshot before restarting their nodes.
+
+## Upgrade Procedure
+
+__Note__: It is assumed you are currently operating a full-node running gaia v2.0.14 with v0.37.15 of the _Cosmos SDK_.
+
+- The version/commit hash of Gaia v2.0.14: `86343f24f39dbc800922e5fb296b77b9d30ebaae`
+
+1. Verify you are currently running the correct version (v2.0.14) of _gaiad_:
+
+   ```bash
+    $ gaiad version --long
+    name: gaia
+    server_name: gaiad
+    client_name: gaiacli
+    version: 2.0.14
+    commit: 86343f24f39dbc800922e5fb296b77b9d30ebaae
+    build_tags: netgo,ledger
+    go: go version go1.15 darwin/amd64
+   ```
+
+2. Make sure your chain halts at the right time and date:
+    February 18, 2021 at 06:00 UTC is in UNIX seconds: `1613628000`
+
+    ```bash
+    perl -i -pe 's/^halt-time =.*/halt-time = 1613628000/' ~/.gaiad/config/app.toml
+    ```
+
+3. Export existing state from `cosmoshub-3`:
+
+   **NOTE**: It is recommended for validators and operators to take a full data snapshot at the export
+   height before proceeding in case the upgrade does not go as planned or if not enough voting power
+   comes online in a sufficient and agreed upon amount of time. In such a case, the chain will fallback
+   to continue operating `cosmoshub-3`. See [Recovery](#recovery) for details on how to proceed.
+
+   Before exporting state via the following command, the `gaiad` binary must be stopped!
+
+   ```bash
+   $ gaiad export --for-zero-height --height=<height> > cosmoshub_3_genesis_export.json
```

Comment:
What's the command/process to obtain the height?

Reply:
That's a good question. Will look for a solution to obtain it after the daemon has stopped.